modules/DGM/PSO.cpp

```diff
@@ -5,11 +5,9 @@
 #include "PSO.h"
 #include "macroses.h"
 
-#include <iterator>
-#include <vector>
 #include <random>
 
-DirectGraphicalModels::PSO::PSO() {
+DirectGraphicalModels::PSO::PSO() : ParamEstAlgorithm(0) {
```

Comment:
Here you make use of _assign_ operator, which is technically not excactly "initialization". You can initialize them as follows:
`DirectGraphicalModels::PSO::PSO() 
    : ParamEstAlgorithm(0)
    , c1(C1_DEFAULT_VALUE)
    , c2(C2_DEFAULT_VALUE)
    , isThreadsEnabled(false)
{}
` 
or directly in the header file